pallets/nft/src/lib.rs

```diff
@@ -481,6 +549,53 @@ pub mod pallet {
 			Ok(().into())
 		}
 
+		/// Transfer an existing NFT asset if it is not listed in an auction
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		/// - `to`: account to transfer the NFT asset to
+		/// - `asset_id`: the asset (class ID, token ID) that will be transferred
+		///
+		/// Emits `TransferedStakcableNft` if successful.
+		#[pallet::weight(T::WeightInfo::transfer_stackable_nft())]
+		#[transactional]
+		pub fn transfer_stackable_nft(
+			origin: OriginFor<T>,
+			to: T::AccountId,
+			asset_id: (ClassIdOf<T>, TokenIdOf<T>),
+			amount: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
+
+			ensure!(
+				Self::check_item_on_listing(asset_id.0, asset_id.1)? == false,
+				Error::<T>::AssetAlreadyInAuction
+			);
+
+			ensure!(
+				amount > Zero::zero()
+					&& Self::get_stackable_collections((asset_id.0, asset_id.1, sender.clone())) >= amount,
+				Error::<T>::InvalidStackableNftTransfer
+			);
+
+			StackableCollections::<T>::try_mutate(
+				(asset_id.0, asset_id.1, sender.clone()),
+				|sender_balance| -> DispatchResult {
+					StackableCollections::<T>::try_mutate(
+						(asset_id.0, asset_id.1, to.clone()),
+						|receiver_balance| -> DispatchResult {
+							*receiver_balance = receiver_balance.saturating_add(amount);
+							*sender_balance = sender_balance.saturating_sub(amount);
+							Self::deposit_event(Event::<T>::TransferedStackableNft(sender, to, asset_id, amount));
+
+							Ok(())
+						},
+					);
+					Ok(())
```

Comment:
Must handle Err or ? in case of an error occur during mutation.

Reply:
I've checked the  possible error cases before the 2 try_mutate closures since I was getting test errors otherwise. Could update that code if needed too ?